src/components/NewDeckForm/NewDeckForm.js

```diff
@@ -0,0 +1,158 @@
+import React, {useState} from "react";
+import {Form, Input, Button, Alert, Checkbox, Select} from "antd";
+import {useDispatch} from "react-redux";
+import {deckTags} from "../../utils/deckTags";
+import {createDeck} from "../../state/actions/decks";
+import styles from "../../components/RecentDecks/RecentDecks.module.css";
+
+const NewDeckForm = props => {
+  const [formValues, setFormValues] = useState({
+    name: "",
+    tags: [],
+    isPublic: false,
+  });
+
+  const dispatch = useDispatch();
+
+  const [state, setState] = useState({
+    success: null,
+  });
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    const numberTags = formValues.tags.map(x => parseInt(x));
+    if (formValues.isPublic === true) {
+      dispatch(createDeck({...formValues, tags: numberTags, isPublic: "1"}))
+        .then(res => {
+          setState({
+            success: true,
+          });
+          setFormValues({
+            name: "",
+            tags: [],
+            isPublic: false,
+          });
+        })
+        .catch(err => {
+          setState({
+            success: false,
+          });
+        });
+    } else if (formValues.isPublic === false) {
+      dispatch(createDeck({...formValues, tags: numberTags, isPublic: "0"}))
+        .then(res => {
+          setState({
+            success: true,
+          });
+          setFormValues({
+            name: "",
+            tags: [],
+            isPublic: false,
+          });
+        })
+        .catch(err => {
+          setState({
+            success: false,
+          });
+        });
+    }
+  };
+
+  const changeCheckbox = e => {
+    setFormValues({...formValues, isPublic: e.target.checked});
+  };
+
+  const handleChange = e => {
+    setFormValues({
+      ...formValues,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  function handleChangeSelection(value) {
+    setFormValues({...formValues, tags: value});
+  }
+
+  const {Option} = Select;
+  const {getFieldDecorator} = props.form;
+  const children = [];
+  const options = deckTags;
+  // eslint-disable-next-line array-callback-return
+  options.map((curr, index) => {
+    children.push(<Option key={index}>{curr}</Option>);
+  });
```

Comment:
This could be a forEach, and you wouldn't need to disable eslint for that line

Reply:
its given by ant design

Reply:
Even though it's given by the example. We can change it. The forEach is the better option